tensorpack/train/trainers.py

```diff
@@ -399,7 +402,7 @@ def allreduce(self, grads):
         with tf.name_scope("HVDAllReduce"):
             for grad, var in grads:
                 if grad is not None:
-                    avg_grad = hvd.allreduce(grad, average=self._average)
+                    avg_grad = hvd.allreduce(grad, average=self._average, compression=self._compression)
```

Comment:
should add a check because this feature is only supported after horovod 0.15

Reply:
Sure, I found it only works after horovod 0.15